coderd/audit.go

```diff
@@ -352,18 +350,34 @@ func auditLogResourceLink(alog database.GetAuditLogsOffsetRow, additionalFields
 		return fmt.Sprintf("/users?filter=%s",
 			alog.ResourceTarget)
 	case database.ResourceTypeWorkspace:
-		workspaceOwner := alog.UserUsername.String
-		if len(additionalFields.WorkspaceOwner) != 0 && additionalFields.WorkspaceOwner != "unknown" {
-			workspaceOwner = additionalFields.WorkspaceOwner
+		workspace, getWorkspaceErr := api.Database.GetWorkspaceByID(ctx, alog.ResourceID)
+		if getWorkspaceErr != nil {
+			return ""
+		}
+		workspaceOwner, getWorkspaceOwnerErr := api.Database.GetUserByID(ctx, workspace.OwnerID)
+		if getWorkspaceOwnerErr != nil {
+			return ""
 		}
 		return fmt.Sprintf("/@%s/%s",
-			workspaceOwner, alog.ResourceTarget)
+			workspaceOwner.Username, alog.ResourceTarget)
 	case database.ResourceTypeWorkspaceBuild:
 		if len(additionalFields.WorkspaceName) == 0 || len(additionalFields.BuildNumber) == 0 {
 			return ""
 		}
+		workspaceBuild, getWorkspaceBuildErr := api.Database.GetWorkspaceBuildByID(ctx, alog.ResourceID)
+		if getWorkspaceBuildErr != nil {
+			return ""
+		}
+		workspace, getWorkspaceErr := api.Database.GetWorkspaceByID(ctx, workspaceBuild.WorkspaceID)
+		if getWorkspaceErr != nil {
+			return ""
+		}
+		workspaceOwner, getWorkspaceOwnerErr := api.Database.GetUserByID(ctx, workspace.OwnerID)
+		if getWorkspaceOwnerErr != nil {
+			return ""
+		}
```

Comment:
I don't love it, but idk how else to get workspace owner. I don't want to pass everything through `additionalFields` because that will not fix old records.